collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -135,21 +137,23 @@ private void scrapeBean(MBeanServerConnection beanConn, ObjectName mbeanName) {
         }
         MBeanAttributeInfo[] attrInfos = info.getAttributes();
 
+        Map<String, MBeanAttributeInfo> name2AttrInfo = new LinkedHashMap<String, MBeanAttributeInfo>();
         for (int idx = 0; idx < attrInfos.length; ++idx) {
             MBeanAttributeInfo attr = attrInfos[idx];
             if (!attr.isReadable()) {
                 logScrape(mbeanName, attr, "not readable");
                 continue;
             }
-
-            Object value;
-            try {
-                value = beanConn.getAttribute(mbeanName, attr.getName());
-            } catch(Exception e) {
-                logScrape(mbeanName, attr, "Fail: " + e);
-                continue;
-            }
-
+            name2AttrInfo.put(attr.getName(), attr);
+        }
+        AttributeList attributes = new AttributeList();
```

Comment:
The object created here isn't used

Reply:
oh - i thought this was setting up an empty collection so that if the getAttributes call fails then we can just log and continue.  then the next block can safely iterate over the empty coolection

Reply:
You can return in the catch